lib/js/src/manager/screen/choiceset/ChoiceCell.js

```diff
@@ -0,0 +1,252 @@
+     */
+    setVoiceCommands (voiceCommands) {
+        this._voiceCommands = voiceCommands;
+        return this;
+    }
+
+    /**
+     * Get the state artwork
+     * @returns {SdlArtwork} - The SdlArtwork
+     */
+    getArtwork () {
+        return this._artwork;
+    }
+
+    /**
+     * Set the state artwork
+     * Maps to Choice.image. Optional image for the cell. This will be uploaded before the cell is
+     * used when the cell is preloaded or presented for the first time.
+     * @param {SdlArtwork} artwork - The SdlArtwork
+     * @returns {ChoiceCell} - A reference to this instance to support method chaining
+     */
+    setArtwork (artwork) {
+        this._artwork = artwork;
+        return this;
+    }
+
+    /**
+     * Get the state secondaryArtwork
+     * @returns {SdlArtwork} - The SdlArtwork
+     */
+    getSecondaryArtwork () {
+        return this._secondaryArtwork;
+    }
+
+    /**
+     * Set the state secondaryArtwork
+     * Maps to Choice.secondaryImage. Optional secondary image for the cell. This will be uploaded
+     * before the cell is used when the cell is preloaded or presented for the first time.
+     * @param {SdlArtwork} secondaryArtwork - The SdlArtwork
+     * @returns {ChoiceCell} - A reference to this instance to support method chaining
+     */
+    setSecondaryArtwork (secondaryArtwork) {
+        this._secondaryArtwork = secondaryArtwork;
+        return this;
+    }
+
+    /**
+     * Get the state choiceId
+     * @private
+     * @returns {Number} - The choice ID
+     */
+    _getChoiceId () {
+        return this._choiceId;
+    }
+
+    /**
+     * Set the state choiceId
+     * @private
+     * @param {Number} choiceId - The choice ID
+     * @returns {ChoiceCell} - A reference to this instance to support method chaining
+     */
+    _setChoiceId (choiceId) {
+        this._choiceId = choiceId;
+        return this;
+    }
+
+    /**
+     * Checks whether two ChoiceCells can be considered equivalent
+     * @param {ChoiceCell} other - The object to compare
+     * @returns {Boolean} - Whether the objects are the same or not
+     */
+    equals (other) {
+        if (other === null || other === undefined) {
+            return false;
+        }
+        if (this === other) {
+            return true;
+        }
+        if (!(other instanceof ChoiceCell)) {
+            return false;
+        }
+        // main comparison check
+        if (this.getText() !== other.getText()) {
+            return false;
+        }
+        if (this.getSecondaryText() !== other.getSecondaryText()) {
+            return false;
+        }
+        if (this.getTertiaryText() !== other.getTertiaryText()) {
+            return false;
+        }
+        if (this.getArtwork() === null && other.getArtwork() !== null) {
+            return false;
+        }
+        if (this.getArtwork() !== null && !this.getArtwork().equals(other.getArtwork())) {
```

Comment:
The .equals() method is part of the SdlFile class, not the SdlArtwork class. Because of that, it won't check the _isTemplate property which is part of the SdlArtwork class.

Reply:
I believe this is the way the Java version does this. Their way of comparing object equality is by implementing the `hashCode`/`equals` method. The SdlArtwork class doesn't implement this but the parent SdlFile does. Unless this is a bug with the Java code as well, it seems to ignore the extra properties SdlArtwork offers as part of the equality check.